### PR TITLE
Closes #2583. Exclude watch-only utxos from coin selection.

### LINKED ISSUE #2583: Shouldn't we check that the coin is spendable here?
https://github.com/zcash/zcash/blob/2a42478bff3e1298f73e0411eb6d69d83b89f6a0/src/wallet/asyncrpcoperation_sendmany.cpp#L805

i.e. that `out.fSpendable == true`
This should usually not be a problem,  unless the user would use a watch only address or something.

It came up for me cause I'm working on a version that doesn't require an input address

### PR DESCRIPTION
**Pull request by bitcartel**

### CODE REVIEW

**Comment by bitcartel** [comment]
I set my editor to strip trailing whitespace... hence the commit being the way it is...

**Comment by arcalinea** [comment]
Rpc test for this?

**Comment by bitcartel** [comment]
@arcalinea RPC test added.

**Comment by bitcartel** [comment]
@zkbot try

**Comment by bitcartel** [comment]
I don' t know how long the tests take on the builder, but I ran (a few times) the updated rpc test locally in a debian 8 vm and it completed successfully.

File: src/wallet/asyncrpcoperation_sendmany.cpp [hunk 1]
```diff
@@ -803,6 +803,10 @@ bool AsyncRPCOperation_sendmany::find_utxos(bool fAcceptCoinbase=false) {
     pwalletMain->AvailableCoins(vecOutputs, false, NULL, true, fAcceptCoinbase);
 
     BOOST_FOREACH(const COutput& out, vecOutputs) {
+        if (!out.fSpendable) {
```
Comment (arielgabizon) [inline src/wallet/asyncrpcoperation_sendmany.cpp]: I'm wondering if in terms of efficiency, we should do these checks of `fSpendable` and `nDepth` inside an altered version of `AvailableCoins`.  Cause right now we are scanning all the outputs twice.

Comment (arielgabizon) [reply src/wallet/asyncrpcoperation_sendmany.cpp]: But I'm fine with doing this optimization later, if we want to fix it like this now.

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**